config/routes.rb

```diff
@@ -123,7 +123,9 @@
     end
   end
 
-  resources :usage, only: [:index]
+  resources :usage, only: [:index] do
+    get :all, action: :all, on: :collection, as: :all, constraints: { format: :csv }
```

Comment:
Format constraints might not work the way you expect here. From the rails docs:

> There is an exception for the format constraint: while it's a method on the Request object, it's also an implicit optional parameter on every path. Segment constraints take precedence and the format constraint is only applied as such when enforced through a hash. For example, get 'foo', constraints: { format: 'json' } will match GET  /foo because the format is optional by default. However, you can use a lambda like in get 'foo', constraints: lambda { |req| req.format == :json } and the route will only match explicit JSON requests.

I'll need to change a couple of the routes that I added recently too 🙈